### DISCUSSION TITLE
Error: relative URL without a base running forge create

### DISCUSSION THREAD

**Discussion by evronm** [foundry-rs/foundry#1459] [Help] [answered]
I can't figure out why I'm getting this error, and I hope someone can help.  The contracts compile fine, but won't deploy.  Here's the command I'm running, and the output:

<code>
$ RUST_BACKTRACE=full forge create --rpc-url 127.0.0.1:8545 --private-key 0xD26aAb4FCB6f76816Fe912A76788c2D14098B6b4 ./src/Factory.sol:Factory
[⠊] Compiling...
No files changed, compilation skipped
Error: 
   0: relative URL without a base

Location:
   cli/src/cmd/create.rs:84

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
   1: __libc_start_main<unknown>
      at /builddir/glibc-2.32/csu/../csu/libc-start.c:314
</code>

I managed to deploy the same contracts using remix, so the problem is in the configuration, not in the code.  And my configuration is the default.

Any help would be greatly appreciated.  I really, really don't want to have to move to truffle or hardhat...

Thanks in advance.

**Comment by onbjerg** [comment, marked as answer]
I'd assume it is because you need to specify the RPC as `http://127.0.0.1:8545`? :thinking:

**Reply by evronm** [reply]
Nope. Tried it both ways, same error.

**Reply by onbjerg** [reply]
What version are you on?

**Reply by evronm** [reply]
forge 0.1.0 (22323e2 2022-03-23T00:09:43.397848305+00:00)

**Reply by onbjerg** [reply]
You're super out of date, run this:

```
curl -L https://foundry.paradigm.xyz | bash
foundryup
```

**Reply by evronm** [reply]
OK, I've upgraded.  I'm now at version 0.2.0.  Slightly different error message:

<code>
RUST_BACKTRACE=full forge create --rpc-url 127.0.0.1:8545 --private-key 0xD26aAb4FCB6f76816Fe912A76788c2D14098B6b4 ./src/Factory.sol:Factory
[⠊] Compiling...
No files changed, compilation skipped
Error: 
   0: relative URL without a base

Location:
   cli/src/cmd/forge/create.rs:125

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
   1: __libc_start_main<unknown>
      at /builddir/glibc-2.32/csu/../csu/libc-start.c:314


</code>

**Reply by onbjerg** [reply]
I'm fairly certain it doesn't work without `http://` in front. Can you try `http://localhost:8545`?

**Reply by evronm** [reply]
OK, we're getting somewhere!  here's what I get now:

<code>
0: (code: -32700, message: The method eth_feeHistory does not exist/is not available, data: None)
</code>

**Reply by evronm** [reply]
BTW, thanks SO much for your help.  Also, should I now mark this as answered and start a different discussion?

**Reply by onbjerg** [reply]
For that last one I think you're trying to communicate with a chain that does not have EIP1559. Try adding the `--legacy` flag :slightly_smiling_face:

**Reply by evronm** [reply]
That did it!  Thanks again.  Love this project!
